ocs_ci/ocs/resources/mcg.py

```diff
@@ -472,45 +473,53 @@ def oc_create_bucketclass(self, name, backingstores, placement):
+    def toggle_aws_bucket_readwrite(self, bucketnames, block=True, wait=True):
         """
         Toggles a bucket's IO using a bucket policy
 
         Args:
-            bucketname: The name of the bucket that should be manipulated
+            bucketnames: A list of bucket names that should be manipulated
             block: Whether to block RW or un-block. True | False
-
-        """
-        if block:
-            bucket_policy = {
-                "Version": "2012-10-17",
-                "Id": "DenyReadWrite",
-                "Statement": [
-                    {
-                        "Effect": "Deny",
-                        "Principal": {
-                            "AWS": "*"
-                        },
-                        "Action": [
-                            "s3:GetObject",
-                            "s3:PutObject",
-                            "s3:ListBucket"
-                        ],
-                        "Resource": [
-                            f"arn:aws:s3:::{bucketname}/*",
-                            f"arn:aws:s3:::{bucketname}"
-                        ]
-                    }
-                ]
-            }
-            bucket_policy = json.dumps(bucket_policy)
-            self.aws_s3_resource.meta.client.put_bucket_policy(
-                Bucket=bucketname, Policy=bucket_policy
-            )
-        else:
-            self.aws_s3_resource.meta.client.delete_bucket_policy(
-                Bucket=bucketname
-            )
+            wait: Whether to wait for the policy change to take effect or not
+
+        """
+        for bucketname in bucketnames:
+            if block:
+                bucket_policy = {
+                    "Version": "2012-10-17",
+                    "Id": "DenyReadWrite",
+                    "Statement": [
+                        {
+                            "Effect": "Deny",
+                            "Principal": {
+                                "AWS": "*"
+                            },
+                            "Action": [
+                                "s3:GetObject",
+                                "s3:PutObject",
+                                "s3:ListBucket"
+                            ],
+                            "Resource": [
+                                f"arn:aws:s3:::{bucketname}/*",
+                                f"arn:aws:s3:::{bucketname}"
+                            ]
+                        }
+                    ]
+                }
+                bucket_policy = json.dumps(bucket_policy)
+                self.aws_s3_resource.meta.client.put_bucket_policy(
+                    Bucket=bucketname, Policy=bucket_policy
+                )
+            else:
+                self.aws_s3_resource.meta.client.delete_bucket_policy(
+                    Bucket=bucketname
+                )
+        if wait:
+            for bucketname in bucketnames:
+                if block:
```

Comment:
you can create a variable and assign it BS_AUTH_FAILED in case of block=True and BS_OPTIMAL in case block=False.

Then you won't need 2 calls to check_backingstore_state

Reply:
created a new variable as suggested